teps/0090-matrix.md

```diff
@@ -721,6 +720,18 @@ If needed, we can also explore providing more granular controls for maximum numb
 or `Runs` from `Matrices` - either at `PipelineRun`, `Pipeline` or `PipelineTask` levels - later.
 This is an option we can pursue after gathering user feedback - it's out of scope for this TEP.
 
+### Failure Strategies
+
+In failure scenarios, the `TaskRuns` or `Runs` created from a `Matrix` will fail fast. That is, when
+any `TaskRun` or `Run` from a given fanned-out `PipelineTask` fails or is cancelled then the other
+`TaskRuns` or `Runs` from the same `PipelineTask` will be cancelled.
```

Comment:
thanks @jerop, looking for more clarification, how are we failing already running `taskRuns` or `runs` from the same `pipelineTask`? Sending termination signal?

Can we implement something like `stopping` mode by default, the similar approach we take when a task in a pipeline fails?

Reply:
the reason I am asking is I am assuming `stopping` is easier to implement 🤣 And does not result in partial execution ...

Reply:
@pritidesai yes, similarly to how we cancel `TaskRuns` for example - it may be easier to implement `stopping` mode but wondering if that's the behavior we want for `Matrix` separate from the implementation - maybe partial execution could be the reason not to do this?

Reply:
`stopping` will be consistent to how we address `failure` of a non-matrix `pipelineTask`.

Just an example, if building an image of one application fails, I do not want that failure to stop building images of 20 other applications.

Reply:
makes sense - added this to the API WG agenda on Monday so we can decide on the best way forward